lib/min_heap.rb

```diff
@@ -44,28 +59,38 @@ def to_s
   end
 
   # This method returns true if the heap is empty
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(0)
+  # Space complexity: O(0)
   def empty?
-    raise NotImplementedError, "Method not implemented yet..."
+    return @store.empty?
   end
 
   private
 
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(logn)
+  # Space complexity: O(0)
   def heap_up(index)
-    
+    while @store[index].key < @store[(index - 1)/2].key
+      swap(index, (index - 1)/2)
+      index = (index - 1)/2
+      return if index == 0
+    end
   end
 
   # This helper method takes an index and 
   #  moves it up the heap if it's smaller
   #  than it's parent node.
   def heap_down(index)
```

Comment:
This isn't working because you've got to:

1.  Check the both left and right child
    1. And swap with the smaller of the two, if needed
2. Consider the case where there is a left child and not a right child